### PR TITLE
Docs: roles logic utils docs

### PR DESCRIPTION
**Pull request by coleFD**

### CODE REVIEW

File: protocols/v2/roles-logic-sv2/src/common_properties.rs [hunk 1]
```diff
@@ -37,8 +42,11 @@ pub trait IsUpstream<Down: IsDownstream, Sel: DownstreamSelector<Down> + ?Sized>
         let check_flags = SetupConnection::check_flags(protocol, self.get_flags(), flags);
         check_version && check_flags
     }
+    /// Should return the channel id
```
Comment (darricksee) [inline protocols/v2/roles-logic-sv2/src/common_properties.rs:45]: Remove the use of the word "Should" here and below.

File: protocols/v2/roles-logic-sv2/src/handlers/job_negotiation.rs [hunk 2]
```diff
@@ -6,22 +6,24 @@ use crate::errors::Error;
 use core::convert::TryInto;
 use job_negotiation_sv2::SetCoinbase;
 
+/// A trait implemented by a downstream to handle SV2 job negotiation messages.
 pub trait ParseServerJobNegotiationMessages
 where
     Self: Sized,
 {
+    // Used to parse job negotiation message and route to the message's respected handler function
```
Comment (darricksee) [inline protocols/v2/roles-logic-sv2/src/handlers/job_negotiation.rs:14]: Was this intentionally `//`

File: protocols/v2/roles-logic-sv2/src/handlers/mining.rs [hunk 3]
```diff
@@ -228,9 +229,10 @@ pub trait ParseUpstreamMiningMessages<
         None
     }
 
+    /// Used to parse and route SV2 mining messages from the upstream based on `message_type` and `payload`
+    /// The implementor of DownstreamMining needs to pass a RequestIdMapper if needing to change the req id.
     /// Proxies likely would want to update a downstream req id to a new one as req id must be
-    /// connection-wide unique
-    /// The implementor of DownstreamMining need to pass a RequestIdMapper if want to change the req id
+    /// connection-wide unique/
```
Comment (darricksee) [inline protocols/v2/roles-logic-sv2/src/handlers/mining.rs:235]: Remove the  trailing `/` -> `.`

File: protocols/v2/roles-logic-sv2/src/handlers/job_negotiation.rs (hunk 2 above)
Comment (coleFD) [reply protocols/v2/roles-logic-sv2/src/handlers/job_negotiation.rs:14]: This was unintentional. The `pub trait` makes those functions public

File: protocols/v2/roles-logic-sv2/src/handlers/job_negotiation.rs [hunk 4]
```diff
@@ -6,22 +6,24 @@ use crate::errors::Error;
...
-            Ok(JobNegotiation::SetCoinbase(message)) => {
-                self_.safe_lock(|x| x.handle_set_coinbase(message)).unwrap()
-            }
+            Ok(JobNegotiation::SetCoinbase(message)) => self_
+                .safe_lock(|x| x.handle_set_coinbase(message))
+                .map_err(|e| crate::Error::PoisonLock(e.to_string()))?,
             Err(e) => Err(e),
         }
     }
+    // Should construct a new coinbase transaction based on the message
```
Comment (darricksee) [inline protocols/v2/roles-logic-sv2/src/handlers/job_negotiation.rs:27]: Same here? Should this be `///` instead. Also, can you remove "Should"?

**Review by darricksee — APPROVED** [review]

**Pull request merged**